druntime/src/object.d

```diff
@@ -642,7 +642,8 @@ class TypeInfo
      */
     size_t getHash(scope const void* p) @trusted nothrow const
     {
-        return hashOf(p);
+        // by default, do not assume anything about the type
+        return 0;
```

Comment:
Anyone have any idea why this would be bad? The default of hashing the pointer seems ludicrous to me -- the pointer comes from anywhere and is a consequence of having a runtime interface vs. a compile time one. It has nothing to do with item equality or hashing.

Reply:
It could be a bad performance regression, not sure if anyone actually relies on it though.